torch/nn/quantized/modules/linear.py

```diff
@@ -171,3 +171,54 @@ def from_float(mod):
         qlinear.out_scale = torch.tensor([act_qparams[0]])
         qlinear.out_zero_point = torch.tensor([act_qparams[1]])
         return qlinear
+
+
+class DynamicLinear(NNLinear):
```

Comment:
I vote for `nn.quantized.dynamic` namespace

Reply:
There is a lot of code that can be reused from nn.quantized.linear : Here is a suggestion: 
# torch/nn/quantized/dynamic/modules/linear.py
`
class Linear(torch.nn.quantized.Linear):
   
    __constants__ = ['bias', 'in_features', 'out_features']

    def __init__(self, in_features, out_features, bias=True):
        super(Linear, self).__init__(in_features, out_features, bias)
 # Remove registration of output scale and zero point
      del out_scale
     del out_zero_point

    def forward(self, x):
        # Write forward for dynamic op
  
   # Inherit methods for load/save from state_dict

    @staticmethod
    def from_float(mod):
        r"""Create a quantized module from a float module or qparams_dict
            Args: `mod` a float module, either produced by torch.quantization utilities
            or directly from user
        """
        assert type(mod) == NNLinear, 'nnq.Linear.from_float only works for nn.Linear'
        assert hasattr(mod, 'qconfig'), 'Input float module must have qconfig defined'
   
        weight_observer = mod.qconfig.weight()
        weight_observer(mod.weight)
        wt_qparams = weight_observer.calculate_qparams()
        bias_scale = (wt_qparams[0] * act_qparams[0]).float()
        qweight = torch.quantize_linear(mod.weight.float(), wt_qparams[0], wt_qparams[1].long().item(), torch.qint8)
        qbias = torch.quantize_linear(mod.bias.float(), bias_scale, 0, torch.qint32)
        qlinear = Linear(mod.in_features, mod.out_features)
        qlinear._packed_weight = torch.ops.quantized.fbgemm_linear_prepack(qweight)
        qlinear.bias = qbias
        return qlinear

Reply:
Updated in https://github.com/pytorch/pytorch/pull/23128.